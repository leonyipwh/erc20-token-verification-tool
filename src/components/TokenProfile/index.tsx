import Typography from '@mui/material/Typography';
import Box from '@mui/material/Box';
import Paper from '@mui/material/Paper';
import Divider from '@mui/material/Divider';

interface IProfileData {
  label: string;
  value: number | string | undefined;
}

const TokenProfile = ({ data }: { data: IProfileData[] }) => {
  return (
    <Paper sx={{ p: 2 }}>
      <Typography
        variant="h6"
        gutterBottom
      >
        Token Profile
      </Typography>
      <Divider />
      {data.map((item, index) => {
        return (
          <Box
            key={index}
            sx={{
              mt: 2
            }}
          >
            <Typography
              variant="subtitle2"
              gutterBottom
              component="span"
            >
              {item.label}:
            </Typography>
            <Typography
              variant="body2"
              gutterBottom
              component="span"
              sx={{ ml: 1 }}
            >
              {item?.value}
            </Typography>
          </Box>
        );
      })}
    </Paper>
  );
};

export default TokenProfile;
