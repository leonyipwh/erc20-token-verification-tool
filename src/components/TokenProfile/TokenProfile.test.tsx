import { render } from '@testing-library/react';
import TokenProfile from './index';

jest.mock('../../helpers/infuraHelper', () => {
  return {
    initProvider: () => {}
  };
});

const profileMock = [
  {
    label: 'Name',
    value: 'BNB'
  },
  {
    label: 'Symbol',
    value: 'BNB'
  },
  {
    label: 'Decimals',
    value: '18'
  },
  {
    label: 'Total Supply',
    value: '16,579,517.05525335'
  }
];

describe('<TokenProfile>', () => {
  it('should render', () => {
    const { asFragment } = render(<TokenProfile data={profileMock} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
