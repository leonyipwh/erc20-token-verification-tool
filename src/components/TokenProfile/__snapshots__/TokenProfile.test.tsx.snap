// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TokenProfile> should render 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-i0bk1x-MuiPaper-root"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
    >
      Token Profile
    </h6>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiBox-root css-1yuhvjn"
    >
      <span
        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-1oxen2w-MuiTypography-root"
      >
        Name:
      </span>
      <span
        class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-yyw3lr-MuiTypography-root"
      >
        BNB
      </span>
    </div>
    <div
      class="MuiBox-root css-1yuhvjn"
    >
      <span
        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-1oxen2w-MuiTypography-root"
      >
        Symbol:
      </span>
      <span
        class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-yyw3lr-MuiTypography-root"
      >
        BNB
      </span>
    </div>
    <div
      class="MuiBox-root css-1yuhvjn"
    >
      <span
        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-1oxen2w-MuiTypography-root"
      >
        Decimals:
      </span>
      <span
        class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-yyw3lr-MuiTypography-root"
      >
        18
      </span>
    </div>
    <div
      class="MuiBox-root css-1yuhvjn"
    >
      <span
        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-1oxen2w-MuiTypography-root"
      >
        Total Supply:
      </span>
      <span
        class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-yyw3lr-MuiTypography-root"
      >
        16,579,517.05525335
      </span>
    </div>
  </div>
</DocumentFragment>
`;
